src/mantine-core/src/components/Select/SelectDropdown/SelectDropdown.tsx

```diff
@@ -33,37 +37,78 @@ export const SelectDropdown = forwardRef<HTMLDivElement, SelectDropdownProps>(
       classNames,
       styles,
       dropdownComponent,
+      referenceElement,
+      direction = 'column',
+      onDirectionChange,
       __staticSelector,
     }: SelectDropdownProps,
     ref
   ) => {
     const { classes } = useStyles(null, { classNames, styles, name: __staticSelector });
 
     return (
-      <Transition
+      <Popper
+        referenceElement={referenceElement}
         mounted={mounted}
         transition={transition}
-        duration={transitionDuration}
-        timingFunction={transitionTimingFunction}
+        transitionDuration={transitionDuration}
+        transitionTimingFunction={transitionTimingFunction}
+        position="bottom"
+        modifiers={[
+          {
+            name: 'flip',
+            enabled: Boolean(onDirectionChange),
+            options: {
+              fallbackPlacements: ['top'],
+            },
+          },
+          {
+            // @ts-ignore
+            name: 'directionControl',
+            enabled: true,
+            phase: 'main',
+            fn: ({ state }) => {
+              const nextDirection = state.placement === 'top' ? 'column-reverse' : 'column';
+
+              if (direction !== nextDirection) {
+                onDirectionChange && onDirectionChange(nextDirection);
+              }
+            },
+          },
+          {
+            // @ts-ignore
```

Comment:
ignores due I couldn't find a way to extend built-in type of `react-popper`

Reply:
That's fine